umbraco/Umbraco.CMS.Backoffice
# Feature: Collection Extension Type

This pull request introduces a new custom element named `umb-collection` to make rendering of collections easier. The `umb-collection `element is designed to align functionality related to collection actions, views, selection, and pagination. 

### Key Features:

Default Extension Kind: The `umb-collection` comes with a default extension kind that empowers various functionalities such as collection actions, views, selection, and pagination. Extenders also have the flexibility to use a custom element and/or APIs as part of their manifest instead of relying solely on the default extension kind.

### **How to Use:**

**Register the Collection:**
Register a collection using the type 'collection'.

```js
const collection = {
    type: 'collection',
    kind: 'default',
    alias: 'My.Collection.Name',
    // Additional configuration properties for the registration
    // ...
};
```

**Registering Actions**
```js
const collectionAction = {
    type: 'collectionAction',
    alias: 'My.CollectionAction.Name',
    // Additional configuration properties for the registration
    // ...
};
```

**Registering Views**
```js
const collectionView = {
    type: 'collectionView',
    // Additional configuration properties for the registration
    // ...
};
```

**Use the Custom Element:**

Incorporate the custom element `umb-collection` in your HTML with an alias attribute.
Ensure that the alias attribute's value matches the one used during the collection registration.

```HTML

<umb-collection alias="My.Collection.Name"></umb-collection>
```

### What to test
* Make sure that a collection is rendered for users. It should include create + invite actions + two collection views
* Make sure that a collection is rendered for user groups. It should include a create action + a table view.
